core/node/fee_model/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -197,36 +329,33 @@ impl GasAdjuster {
                     * BLOB_GAS_PER_BYTE as f64
                     * self.config.internal_pubdata_pricing_multiplier;
 
-                self.bound_blob_base_fee(calculated_price)
-            }
-            PubdataSendingMode::Calldata => {
-                self.estimate_effective_gas_price() * self.pubdata_byte_gas()
+                self.bound_pubdata_fee(calculated_price)
             }
+            PubdataSendingMode::Calldata => self.bound_pubdata_fee(
+                (self.estimate_effective_gas_price() * L1_GAS_PER_PUBDATA_BYTE as u64) as f64,
+            ),
             PubdataSendingMode::Custom => {
                 // Fix this when we have a better understanding of dynamic pricing for custom DA layers.
                 // GitHub issue: https://github.com/matter-labs/zksync-era/issues/2105
                 0
             }
+            PubdataSendingMode::RelayedL2Calldata => {
+                self.bound_pubdata_fee(self.l2_pubdata_price_statistics.median().as_u64() as f64)
+            }
         }
     }
 
-    fn pubdata_byte_gas(&self) -> u64 {
-        match self.commitment_mode {
-            L1BatchCommitmentMode::Validium => 0,
-            L1BatchCommitmentMode::Rollup => L1_GAS_PER_PUBDATA_BYTE.into(),
-        }
-    }
-
-    fn bound_blob_base_fee(&self, blob_base_fee: f64) -> u64 {
+    fn bound_pubdata_fee(&self, pubdata_fee: f64) -> u64 {
```

Comment:
Grammar (?): I understand that it's used in multiple other places, but it seem that the fee is "bounded" (by certain numerical limits) rather than "bound" (past participle of "bind"; i.e. tied / connected to something).

Reply:
I will rename to `cap_pubdata_fee`